domain/src/main/java/org/oppia/domain/oppialogger/loguploader/LogUploadWorkRequest.kt

```diff
@@ -0,0 +1,80 @@
+package org.oppia.domain.oppialogger.loguploader
+
+import android.content.Context
+import androidx.work.Constraints
+import androidx.work.Data
+import androidx.work.ExistingPeriodicWorkPolicy
+import androidx.work.NetworkType
+import androidx.work.PeriodicWorkRequest
+import androidx.work.WorkManager
+import org.oppia.domain.oppialogger.ApplicationStartupListener
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+import javax.inject.Singleton
+
+private const val OPPIA_EXCEPTION_WORK = "OPPIA_EXCEPTION_WORK_REQUEST"
+private const val OPPIA_EVENT_WORK = "OPPIA_EVENT_WORK_REQUEST"
+
+/** Enqueues unique periodic work requests for uploading events and exceptions to the remote service on application creation. */
+@Singleton
+class LogUploadWorkRequest @Inject constructor(
+  private val context: Context
+) : ApplicationStartupListener {
+
+  private val workerConstraints = Constraints.Builder()
+    .setRequiredNetworkType(NetworkType.CONNECTED)
+    .setRequiresBatteryNotLow(true)
+    .setRequiresStorageNotLow(true)
+    .build()
+
+  override fun onCreate() {
+    val workManager = WorkManager.getInstance(context)
+    enqueueWorkRequestForEvents(workManager)
+    enqueueWorkRequestForExceptions(workManager)
+  }
+
+  /** Enqueues a unique periodic work request for uploading events to the remote service. */
+  private fun enqueueWorkRequestForEvents(workManager: WorkManager) {
+    val workerCase =
+      Data.Builder()
+        .putString(
+          LogUploadWorker.WORKER_CASE_KEY,
```

Comment:
What purpose does this enum to string conversion serve? In general that’s not a great practice since we can’t rely on enum constants to be a particular value (at least in Java).

Reply:
Okay so the input data for the worker works as a bundle. It only deals in int, string, float, double, etc. values. Now I need  to handle two cases(event and exception) inside the doWork() of the worker. So to handle both these things I'm doing this way. Should I go for something else?

Reply:
Modified, PTAL.